scripts/publish_posthook.ts

```diff
@@ -11,6 +12,29 @@ if (args.length != 4) {
 	process.exit(1);
 }
 
+// copyDeclarationFilesToDestination copies the declaration files contained in
+// `src` into `dest` with their respective subfolder structure.
+// e.g.:
+//              src/x/y/z.d.ts -> dist/x/y/z.d.ts.
+function copyDeclarationFilesToDestination(src: string, dest: string) {
+	const options: WalkOptions = {
+		listeners: {
+			file: (base, stats, next) => {
+				if (!stats.name.includes(".d.ts")) return next();
+
+				const relativeSrc = [base, stats.name].join("/");
+
+				const strippedBase = base.slice(src.length + 1); // +1 => remove leading "/".
+				const relativeDest = [dest, strippedBase, stats.name].join("/");
+				copyFileSync(relativeSrc, relativeDest);
+
+				next();
```

Comment:
not `return`? because in the first check you return it...

Reply:
I just did it like this because I wanted it to be a one-liner.

```ts
if (...) return next();
```

instead of

```ts
if (...) {
    next();
    return;
}
```

Reply:
nifty 😎